server/handlers.go

```diff
@@ -555,16 +555,25 @@ func getURL(r *http.Request) *url.URL {
 		u.Scheme = "http"
 	}
 
-	if u.Host != "" {
-	} else if host, port, err := net.SplitHostPort(r.Host); err != nil {
-		u.Host = r.Host
-	} else {
-		if port == "80" && u.Scheme == "http" {
-			u.Host = host
-		} else if port == "443" && u.Scheme == "https" {
+	if u.Host == "" {
```

Comment:
I should test it, but this will work only in case of tcp proxy/firewall where no `Host` is set. In case of an http proxy where we set the `Host` header the `--proxy-port` option would be ignored

Reply:
or in general if the client send an `Host` header

Reply:
In my testing, `u.Host` was _always_ empty regardless of how I sent the request to the server.

After further investigation, `r.URL.String()` was always either an empty string or simply "/" 100% of the time, regardless of what I was doing. This lead to the Host never being parsed into `u` by default regardless. The only place where the host seems to be defined is on `r.Host`, but NOT on `r.URL`

If you can find a case where `r.URL.String()` would _ever_ have host information, I would be curious, as I could never find a case where that was true. I don't think that the gorilla mux router includes host information on the URL of *http.Request that it passes into handlers; I believe it only provides host information on the *http.Request struct itself, but I could be wrong there.

The only reason I left in that if statement was because that was the behavior beforehand, although I currently don't believe that's ever actually used.

Reply:
@cheeseandcereal I'm going to do some test next week.
what kind of test did you try?

Reply:
I tried sending http requests to the server doing all of the following

- submitting requests with no Host header
- submitting requests with a normal Host header
- submitting requests with a Host header with a combined port (example.com:1234)
- doing all of the above on both the http and tls listeners

None of those cases resulted in host information on `r.URL`